### PR TITLE
Assigning 412 Precondition failed to dlete produc with product associated

### PR DESCRIPTION
**Pull request by jmmovilla**
**** REVIEWERS *****
@jesuspg

### CODE REVIEW

**Comment by jesuspg** [comment]
LGTM

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/exception/ErrorCode.java [hunk 1]
```diff
@@ -39,7 +39,8 @@
             "The environment is being used by an instance",
             "(.*)InvalidEntityException: (.*)is being used(.*)",
             403),
-    NAME_NO_VALID(41, "The entity is not valid", "(.*)InvalidEntityException:(.*)", 400),
+    PRODUCTRELEASES_ASSOCIATED(41, "This product has product releases associated", "(.*)InvalidEntityException:(.*)releases associated(.*)", 412),
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/exception/ErrorCode.java]: more than 120 characters?

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/resources/ProductResourceImpl.java [hunk 2]
```diff
@@ -481,8 +481,9 @@ public void delete(String name) throws APIException {
         	log.warning("EntityNotFoundException: " + e.getMessage());
         	throw new APIException(new EntityNotFoundException(Product.class,name, e));
         } catch (InvalidEntityException e2) {
-        	log.warning("InvalidEntityException: " + e2.getMessage());
-        	throw new APIException(new InvalidEntityException(Product.class, e2));
+        	String mes = "The product " + name + " has releases associated. ";
+        	log.warning("InvalidEntityException: " + mes +  e2.getMessage());
+        	throw new APIException(new InvalidEntityException(Product.class, new Exception(mes)));
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/resources/ProductResourceImpl.java]: Just to know, how to link to the variable PRODUCTRELEASES_ASSOCIATED? I can imagine it but I would like to confirm.

Comment (jmmovilla) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/resources/ProductResourceImpl.java]: Linked via the log message

Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/resources/ProductResourceImpl.java]: NTC

**Comment by flopezag** [comment]
LGTM

**Pull request merged**